apps/prairielearn/src/pages/instructorAssessmentAccessOverrides/instructorAssessmentAccessOverrides.js

```diff
@@ -0,0 +1,196 @@
+    res.render(__filename.replace(/\.js$/, '.ejs'), res.locals);
+  }),
+);
+
+router.post(
+  '/',
+  asyncHandler(async (req, res) => {
+    if (!res.locals.authz_data.has_course_instance_permission_edit) {
+      throw error.make(403, 'Access denied (must be a student data editor)');
+    }
+    if (!res.locals.assessment_access_overrides_enabled) {
+      throw error.make(403, 'Access denied (feature not available)');
+    }
+
+    if (req.body.__action === 'add_new_override') {
+      const { user_id, group_id } = await getUserOrGroupId({
+        course_instance_id: res.locals.course_instance.id,
+        assessment: res.locals.assessment,
+        uid: req.body.student_uid,
+        group_name: req.body.group_name,
+      });
+      await runInTransactionAsync(async () => {
+        const inserted = await sqldb.queryOneRowAsync(sql.insert_assessment_access_policy, {
+          assessment_id: res.locals.assessment.id,
+          created_by: res.locals.authn_user.user_id,
+          credit: req.body.credit,
+          end_date: new Date(req.body.end_date),
+          group_name: req.body.group_name || null,
+          note: req.body.note || null,
+          start_date: new Date(req.body.start_date),
+          group_id: group_id || null,
+          user_id: user_id || null,
+          timezone: res.locals.course_instance.display_timezone,
+        });
+        await insertAuditLog({
+          authn_user_id: res.locals.authn_user.user_id,
+          table_name: 'assessment_access_policies',
+          action: 'insert',
+          institution_id: res.locals.institution.id,
+          course_id: res.locals.course.id,
+          course_instance_id: res.locals.course_instance.id,
+          new_state: JSON.stringify(inserted.rows[0]),
+          row_id: inserted.rows[0].id,
+        });
+      });
+
+      res.redirect(req.originalUrl);
+    } else if (req.body.__action === 'delete_override') {
+      await runInTransactionAsync(async () => {
+        const deletedAccessPolicy = await sqldb.queryZeroOrOneRowAsync(
+          sql.delete_assessment_access_policy,
+          {
+            assessment_id: res.locals.assessment.id,
+            unsafe_assessment_access_policies_id: req.body.policy_id,
+          },
+        );
+        if (deletedAccessPolicy.rows.length > 0) {
+          await insertAuditLog({
+            authn_user_id: res.locals.authn_user.user_id,
+            table_name: 'assessment_access_policies',
+            action: 'delete',
+            institution_id: res.locals.institution.id,
+            course_id: res.locals.course.id,
+            course_instance_id: res.locals.course_instance.id,
+            old_state: JSON.stringify(deletedAccessPolicy.rows[0]),
+            row_id: deletedAccessPolicy.rows[0].id,
+          });
+        }
+      });
+
+      res.redirect(req.originalUrl);
+    } else if (req.body.__action === 'edit_override') {
+      // const timezone = res.locals.course_instance.display_timezone;
+
```

Comment:
If this is not needed, please delete it.

Reply:
comment deleted in https://github.com/PrairieLearn/PrairieLearn/pull/9103/commits/4e9b3485e4e3ad2ffdec485fd84e9483b7ae89e8